File: RadialMenu/RadialMenuNet/StyleFactory.cs
using System.Drawing;
using SpaceVIL;
using SpaceVIL.Common;
using SpaceVIL.Decorations;
using SpaceVIL.Core;

namespace RadialMenu
{
    public static class StyleFactory
    {
        public static readonly Color CommonBackground = Color.FromArgb(60, 60, 60);
        public static readonly Color ContactBackground = Color.FromArgb(55, 55, 55);

        public static Style GetRadialMenuDefaultStyle()
        {
            Style style = Style.GetDefaultCommonStyle();

            style.SetSizePolicy(SizePolicy.Expand, SizePolicy.Expand);
            style.SetBackground(0, 0, 0, 120);
            style.IsVisible = false;

            Style hideStyle = Style.GetButtonCoreStyle();
            hideStyle.SetPadding(15, 20, 15, 5);
            hideStyle.SetBackground(150, 150, 150);
            hideStyle.SetSize(60, 60);
            hideStyle.SetBorder(new Border(Color.Transparent, new CornerRadius(30), 0));
            hideStyle.SetShadow(new Shadow(5, 3, 3, Color.Black));
            hideStyle.IsShadowDrop = true;
            style.AddInnerStyle("hidebutton", hideStyle);

            Style addStyle = Style.GetButtonCoreStyle();
            addStyle.Font = DefaultsService.GetDefaultFont(FontStyle.Bold, 12);
            addStyle.SetBackground(100, 200, 130);
            addStyle.SetSize(50, 30);
            addStyle.SetBorder(new Border(Color.Transparent, new CornerRadius(15), 0));
            addStyle.SetShadow(new Shadow(5, 3, 3, Color.Black));
            addStyle.IsShadowDrop = true;
            ItemState hover = new ItemState(Color.FromArgb(30, 255, 255, 255));
            hover.Border = new Border(Color.White, new CornerRadius(15), 2);
            addStyle.AddItemState(ItemStateType.Hovered, hover);
            style.AddInnerStyle("addbutton", addStyle);

            return style;
        }

        public static Style GetContactFaceStyle()
        {
            Style style = Style.GetButtonCoreStyle();
            style.SetBackground(120, 120, 120);
            style.SetBackground(5, 162, 232);
            style.SetAlignment(ItemAlignment.HCenter, ItemAlignment.VCenter);
            style.SetPadding(8, 8, 8, 8);
            style.Shape = GraphicsMathService.GetEllipse(30, 30);
            style.AddItemState(ItemStateType.Hovered, new ItemState(Color.FromArgb(60, 255, 255, 255)));
            style.AddItemState(ItemStateType.Pressed, new ItemState(Color.FromArgb(60, 0, 0, 0)));

            return style;
        }

        public static Style GetMenuItemStyle()
        {
            Style style = Style.GetMenuItemStyle();
            style.BorderRadius = new CornerRadius(3);
            style.SetForeground(32, 32, 32);
            style.AddItemState(ItemStateType.Hovered, new ItemState(Color.FromArgb(30, 0, 0, 0)));
            return style;
        }

        public static Style GetMenuStyle()
        {
            Style style = Style.GetContextMenuStyle();
            style.SetBackground(210, 210, 210);
            style.BorderRadius = new CornerRadius(5);
            style.SetShadow(new Shadow(5, 2, 2, Color.FromArgb(180, 0, 0, 0)));
            return style;
        }

        public static Style GetRoundedButtonStyle()
        {
            Style style = Style.GetButtonCoreStyle();
            style.SetAlignment(ItemAlignment.VCenter, ItemAlignment.HCenter);
            style.SetSize(50, 50);
            style.SetShadow(new Shadow(8, 0, 0, Color.Black));
            style.IsShadowDrop = true;
            style.SetBorder(new Border(Color.Transparent, new CornerRadius(25), 0));
            return style;
        }

        public static Style GetContactStyle()
        {
            Style style = Style.GetDefaultCommonStyle();
            style.SetSizePolicy(SizePolicy.Fixed, SizePolicy.Fixed);
            style.Background = Color.Transparent;

            Style layout = Style.GetVerticalStackStyle();
            layout.SetSpacing(0, 5);
            style.AddInnerStyle("layout", layout);

            Style face = StyleFactory.GetContactFaceStyle();
            style.AddInnerStyle("face", face);

            Style name = Style.GetLabelStyle();
            name.Font = DefaultsService.GetDefaultFont(12);
            name.Background = StyleFactory.ContactBackground;
            name.SetSizePolicy(SizePolicy.Fixed, SizePolicy.Fixed);
            name.SetSize(26, 26);
            name.BorderRadius = new CornerRadius(name.Width / 2);
            name.SetAlignment(ItemAlignment.HCenter);
            name.SetTextAlignment(ItemAlignment.HCenter, ItemAlignment.VCenter);
            name.SetShadow(new Shadow(5, 0, 0, Color.Black));
            name.IsShadowDrop = true;
            style.AddInnerStyle("name", name);

            Style notification = Style.GetLabelStyle();
            notification.SetSizePolicy(SizePolicy.Fixed, SizePolicy.Fixed);
            notification.SetTextAlignment(ItemAlignment.HCenter, ItemAlignment.VCenter);
            notification.SetAlignment(ItemAlignment.Right, ItemAlignment.Top);
            notification.SetSize(20, 20);
            notification.SetBorder(new Border(Color.White, new CornerRadius(10), 1));
            notification.Background = StyleFactory.ContactBackground;
            notification.Font = DefaultsService.GetDefaultFont(FontStyle.Bold, 12);
            style.AddInnerStyle("notification", notification);

            return style;
        }
    }
}

File: Apps/Demos/RadialMenu/RadialMenuJVM/src/main/java/RadialMenu/StyleFactory.java
package RadialMenu;

import java.awt.Color;
import java.awt.Font;

import com.spvessel.spacevil.GraphicsMathService;
import com.spvessel.spacevil.Common.DefaultsService;
import com.spvessel.spacevil.Decorations.Border;
import com.spvessel.spacevil.Decorations.CornerRadius;
import com.spvessel.spacevil.Decorations.ItemState;
import com.spvessel.spacevil.Decorations.Shadow;
import com.spvessel.spacevil.Decorations.Style;
import com.spvessel.spacevil.Flags.ItemAlignment;
import com.spvessel.spacevil.Flags.ItemStateType;
import com.spvessel.spacevil.Flags.SizePolicy;

public class StyleFactory {
    public static final Color commonBackground = new Color(60, 60, 60);
    public static final Color contactBackground = new Color(55, 55, 55);
    public static final Color transparent = new Color(0, 0, 0, 0);

    private StyleFactory() {
    }

    public static Style getMenuItemStyle() {
        Style style = Style.getMenuItemStyle();
        style.borderRadius = new CornerRadius(3);
        style.setForeground(32, 32, 32);
        style.addItemState(ItemStateType.HOVERED, new ItemState(new Color(0, 0, 0, 30)));
        return style;
    }
    
    public static Style getMenuStyle() {
        Style style = Style.getContextMenuStyle();
        style.setBackground(210, 210, 210);
        style.borderRadius = new CornerRadius(5);
        style.setShadow(new Shadow(5, 2, 2, new Color(0, 0, 0, 180)));
        return style;
    }

    public static Style getRoundedButtonStyle() {
        Style style = Style.getButtonCoreStyle();
        style.setAlignment(ItemAlignment.VCENTER, ItemAlignment.HCENTER);
        style.setSize(50, 50);
        style.setShadow(new Shadow(8, 0, 0, Color.BLACK));
        style.isShadowDrop = true;
        style.setBorder(new Border(new Color(0, 0, 0, 0), new CornerRadius(25), 0));
        return style;
    }

    public static Style getRadialMenuDefaultStyle() {
        Style style = Style.getDefaultCommonStyle();

        style.setSizePolicy(SizePolicy.EXPAND, SizePolicy.EXPAND);
        style.setBackground(0, 0, 0, 120);
        style.isVisible = false;

        Style hideStyle = Style.getButtonCoreStyle();
        hideStyle.setPadding(15, 20, 15, 5);
        hideStyle.setBackground(150, 150, 150);
        hideStyle.setSize(60, 60);
        hideStyle.setBorder(new Border(new Color(0, 0, 0, 0), new CornerRadius(30), 0));
        hideStyle.setShadow(new Shadow(5, 3, 3, Color.BLACK));
        hideStyle.isShadowDrop = true;
        style.addInnerStyle("hidebutton", hideStyle);

        Style addStyle = Style.getButtonCoreStyle();
        addStyle.font = DefaultsService.getDefaultFont(Font.BOLD, 12);
        addStyle.setBackground(100, 200, 130);
        addStyle.setSize(50, 30);
        addStyle.setBorder(new Border(new Color(0, 0, 0, 0), new CornerRadius(15), 0));
        addStyle.setShadow(new Shadow(5, 3, 3, Color.BLACK));
        addStyle.isShadowDrop = true;
        ItemState hover = new ItemState(new Color(255, 255, 255, 30));
        hover.border = new Border(Color.WHITE, new CornerRadius(15), 2);
        addStyle.addItemState(ItemStateType.HOVERED, hover);
        style.addInnerStyle("addbutton", addStyle);

        return style;
    }

    public static Style getContactStyle() {
        Style style = Style.getDefaultCommonStyle();
        style.setSizePolicy(SizePolicy.FIXED, SizePolicy.FIXED);
        style.background = StyleFactory.transparent;

        Style layout = Style.getVerticalStackStyle();
        layout.setSpacing(0, 5);
        style.addInnerStyle("layout", layout);

        Style face = Style.getButtonCoreStyle();
        face.setSizePolicy(SizePolicy.FIXED, SizePolicy.FIXED);
        face.setShadow(new Shadow(5, 2, 2, Color.BLACK));
        face.isShadowDrop = true;
        face.setBackground(120, 120, 120);
        face.setBackground(5, 162, 232);
        face.setAlignment(ItemAlignment.HCENTER, ItemAlignment.VCENTER);
        face.setPadding(8, 8, 8, 8);
        face.shape = GraphicsMathService.getEllipse(30, 30);
        face.addItemState(ItemStateType.HOVERED, new ItemState(new Color(255, 255, 255, 60)));
        face.addItemState(ItemStateType.PRESSED, new ItemState(new Color(0, 0, 0, 60)));
        style.addInnerStyle("face", face);

        Style name = Style.getLabelStyle();
        name.font = DefaultsService.getDefaultFont(12);
        name.background = StyleFactory.contactBackground;
        name.setSizePolicy(SizePolicy.FIXED, SizePolicy.FIXED);
        name.setSize(26, 26);
        name.borderRadius = new CornerRadius(name.width / 2);
        name.setAlignment(ItemAlignment.HCENTER);
        name.setTextAlignment(ItemAlignment.HCENTER, ItemAlignment.VCENTER);
        name.setShadow(new Shadow(5, 0, 0, Color.BLACK));
        name.isShadowDrop = true;
        style.addInnerStyle("name", name);

        Style notification = Style.getLabelStyle();
        notification.setSizePolicy(SizePolicy.FIXED, SizePolicy.FIXED);
        notification.setTextAlignment(ItemAlignment.HCENTER, ItemAlignment.VCENTER);
        notification.setAlignment(ItemAlignment.RIGHT, ItemAlignment.TOP);
        notification.setSize(20, 20);
        notification.setBorder(new Border(Color.WHITE, new CornerRadius(10), 1));
        notification.background = StyleFactory.contactBackground;
        notification.font = DefaultsService.getDefaultFont(Font.BOLD, 12);
        style.addInnerStyle("notification", notification);

        return style;
    }
}